dev-tools/mage/build.go

```diff
@@ -151,6 +190,12 @@ func Build(params BuildArgs) error {
 	if params.CGO {
 		cgoEnabled = "1"
 	}
+
+	if FIPSBuild {
+		cgoEnabled = "1"
+		env["GOEXPERIMENT"] = "systemcrypto"
```

Comment:
I think this would be more visible if we set it using a build tag instead. According to https://pkg.go.dev/internal/goexperiment, the tag would be `goexperiment.systemcrypto`.

Reply:
The alternative of using build tags is also mentioned in https://github.com/microsoft/go/blob/microsoft/main/eng/doc/fips/README.md#build-tags. Functionality wise either approach should be valid.

Reply:
I tried changing this to add the build tag, but it changed the behaviour of the build command.

Currently if you run `FIPS=true mage build` without microsoft/go, the build fails.
Changing from an env var to another build tag allows it to succeed.
I think having a failure here is desirable as it functions as another check to ensure the binary is built correctly